odoo/tools/misc.py

```diff
@@ -1254,13 +1254,55 @@ def __gt__(self, other): return self.val < other.val
     def __le__(self, other): return self.val >= other.val
     def __lt__(self, other): return self.val > other.val
 
+# there is contextlib.suppress since py3
 @contextmanager
 def ignore(*exc):
     try:
         yield
     except exc:
         pass
 
+class reraise_x_as(ContextDecorator):
+    """
+    Hide some exceptions behind a fake alternative error. Can be used as
```

Comment:
Not fake. When the ORM converts a `KeyError` raised by the cache to a `MissingError`, `MissingError` is not *fake*.